test/session_pool/session_pool.test.js

```diff
@@ -100,6 +100,31 @@ describe('SessionPool - testing session pool', () => {
             expect(session.sessionPool).toEqual(sessionPool);
         });
 
+        test('should retrieve non-random session by ID', async () => {
+            const sessionId = 'testId';
+            const anotherSessionId = 'anotherTestId';
+            expect(sessionPool.sessions.length).toBe(0);
+            expect(sessionPool.usableSessionsCount).toBe(0);
+
+            const generatedSession = await sessionPool.getSession(sessionId);
+            generatedSession.userData = { test: sessionId };
+            expect(sessionPool.sessions.length).toBe(1);
+            expect(sessionPool.usableSessionsCount).toBe(1);
+            expect(generatedSession.id).toEqual(sessionId); // eslint-disable-line
+            expect(generatedSession.userData.test).toEqual(sessionId); // eslint-disable-line
```

Comment:
Here we're using a`get` function to `create` a session in the pool. This is not a good idea. We should come up with a different interface to do this. `addSession` or `newSession` could work.

Reply:
Actually that was my thought exactly at some point but then using one method and not caring about what's going on in the background seemed more convenient from a users' standpoint. I was thinking `createSession` originally :-)